### DISCUSSION TITLE
Issues with embeddables set with `prefix: false`.

### DISCUSSION THREAD

**Discussion by nicomouss** [mikro-orm/mikro-orm#5066] [General]
### Describe the bug

In mikro-orm v6, there seems to be inconsistencies regarding embeddables when `prefix` is set to `false`.

Please check reproduction code provided below.

### Reproduction

```ts
class Place {
    id: number;
    name: string;
    altitude: Altitude;
    population: Population;
}

class Altitude {
    value1: number;
    value2: number;
}

class Population {
    value1: number;
    value2: number;
}

const placeSchema = new EntitySchema<Place>({
    class: Place,
    properties: {
        id: { primary: true, type: Number },
        name: { type: String },
        altitude: {
            kind: "embedded",
            entity: () => Altitude,
            prefix: false
        },
        population: {
            kind: "embedded",
            entity: () => Population,
            prefix: false
        },
    },
    tableName: "Place",
});

const altitudeSchema = new EntitySchema<Altitude>({
    class: Altitude,
    embeddable: true,
    properties: {
        value1: {
            fieldName: "altitude1",
            type: Number,
        },
        value2: {
            fieldName: "altitude2",
            type: Number,
        },
    },
});

const populationSchema = new EntitySchema<Population>({
    class: Population,
    embeddable: true,
    properties: {
        value1: {
            fieldName: "population1",
            type: Number,
        },
        value2: {
            fieldName: "population2",
            type: Number,
        },
    },
});
```

This produces the following DDL:

```ts
create table "Place" (
  "id" serial primary key, 
  "name" varchar(255) not null, 
  "population1" int not null, 
  "population2" int not null
);
```

But it should produce:

```ts
create table "Place" (
  "id" serial primary key, 
  "name" varchar(255) not null, 
  "altitude1" int not null, 
  "altitude2" int not null,
  "population1" int not null, 
  "population2" int not null
);
```

Also an exception is thrown in case the `fieldName` of a property of the embedded entity is set with the same name as the parent property holding that embedded entity (see `fieldName` set below with `altitude` and `population`, which are also the names of the properties in the parent `Place` entity):

```ts
class Place {
    id: number;
    name: string;
    altitude: Altitude;
    population: Population;
}

class Altitude {
    value1: number;
    value2: number;
}

class Population {
    value1: number;
    value2: number;
}

const placeSchema = new EntitySchema<Place>({
    class: Place,
    properties: {
        id: { primary: true, type: Number },
        name: { type: String },
        altitude: {
            kind: "embedded",
            entity: () => Altitude,
            prefix: false
        },
        population: {
            kind: "embedded",
            entity: () => Population,
            prefix: false
        },
    },
    tableName: "Place",
});

const altitudeSchema = new EntitySchema<Altitude>({
    class: Altitude,
    embeddable: true,
    properties: {
        value1: {
            fieldName: "altitude",
            type: Number,
        },
        value2: {
            fieldName: "altitude2",
            type: Number,
        },
    },
});

const populationSchema = new EntitySchema<Population>({
    class: Population,
    embeddable: true,
    properties: {
        value1: {
            fieldName: "population",
            type: Number,
        },
        value2: {
            fieldName: "population2",
            type: Number,
        },
    },
});
```

Exception thrown:
```
MetadataError: Duplicate fieldNames are not allowed: Place.population (fieldName: 'population'), Place.value1 (fieldName: 'value1')
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

v6

### Node.js version

18

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Got myself confused with the repro (as it felt like you want polymorphism which requires a different entity definition), but now I see it is probably something to fix... Will take a closer look, I know where the problem originates, and might be an easy fix.

**Comment by B4nan** [comment]
This should help with both: b43ef6334b28cf8a128260886c3453bac8c510ff

**Reply by nicomouss** [reply]
Yes, just tested out, it indeed fixes the problems !
Very nice, thanks !
